packages/grant-explorer/src/features/round/ViewCartPage/SummaryContainer.tsx

```diff
@@ -0,0 +1,392 @@
+    ChainId[]
+  >(Object.keys(projectsByChain).map(Number));
+
+  const [checkedOutChainIds, setCheckedOutChainIds] = useState<ChainId[]>([]);
+
+  /** The ID of the current chain (from wallet) */
+  const currentChainId = (chain?.id ?? chainIdsBeingCheckedOut[0]) as ChainId;
+  const { data: signer } = useSigner();
+
+  const { switchNetworkAsync } = useSwitchNetwork({
+    onSuccess: () => {
+      setChainSwitched(true);
+    },
+  });
+
+  /** The current payout token  */
+  const currentPayoutToken = payoutTokens[currentChainId];
+
+  /** We find the round that ends last, and take its end date as the permit deadline */
+  const currentPermitDeadline =
+    rounds
+      ?.sort((a, b) => a.roundEndTime.getTime() - b.roundEndTime.getTime())[0]
+      .roundEndTime.getTime() ?? 0;
+
+  const totalDonationsPerChain = useMemo(() => {
+    return Object.fromEntries(
+      Object.entries(projectsByChain).map(([key, value]) => [
+        Number(key) as ChainId,
+        value
+          .map((project) => project.amount)
+          .reduce(
+            (acc, amount) =>
+              acc.add(
+                ethers.utils.parseUnits(
+                  amount ? amount : "0",
+                  payoutTokens[Number(key) as ChainId].decimal
+                )
+              ),
+            BigNumber.from(0)
+          ),
+      ])
+    );
+  }, [payoutTokens, projectsByChain]);
+
+  const navigate = useNavigate();
+  const { address } = useAccount();
+
+  const [insufficientBalance, setInsufficientBalance] = useState(false);
+  const [transactionReplaced, setTransactionReplaced] = useState(false);
+  const [emptyInput, setEmptyInput] = useState(false);
+
+  const [openChainConfirmationModal, setOpenChainConfirmationModal] =
+    useState(false);
+  const [openMRCProgressModal, setOpenMRCProgressModal] = useState(false);
+  const [openErrorModal, setOpenErrorModal] = useState(false);
+  const [errorModalSubHeading, setErrorModalSubHeading] = useState<
+    string | undefined
+  >();
+  /* Donate without matching warning modal */
+  const [donateWarningModalOpen, setDonateWarningModalOpen] = useState(false);
+
+  const [chainSwitched, setChainSwitched] = useState(false);
+
+  const {
+    submitDonations,
+    tokenApprovalStatus,
+    voteStatus,
+    indexingStatus,
+    txHash,
+    setTokenApprovalStatus,
+    setVoteStatus,
+  } = useQFDonation();
+
+  /** This useffect is the main "driver" of state of this component */
+  useEffect(() => {
+    async function run() {
+      if (
+        tokenApprovalStatus === ProgressStatus.IS_ERROR ||
+        voteStatus === ProgressStatus.IS_ERROR
+      ) {
+        setTimeout(() => {
+          setOpenMRCProgressModal(false);
+          // if (transactionReplaced) {
+          //   setErrorModalSubHeading("Transaction cancelled. Please try again.");
+          // }
+          setOpenErrorModal(true);
+        }, modalDelayMs);
+      }
+
+      if (indexingStatus === ProgressStatus.IS_ERROR) {
```

Comment:
Are these if's mutually exclusive or any combination of them is meant to trigger at the same time? If they're mutually exclusive, please replace them with `else if` to avoid accidentally triggering a combination.

Reply:
Good point but outdated.